src/test/java/com/devonfw/sample/archunit/NamingConventionTest.java

```diff
@@ -37,5 +49,88 @@ public class NamingConventionTest {
                     .that().areAssignableTo(AbstractEto.class)
                     .should().haveSimpleNameEndingWith("Eto")
                     .because("Classes extending AbstractEto must follow the naming convention by ending with 'Eto'.");
+    /**
+     * DevonApplicationPersistenceEntityCheck verifying that classes extending ApplicationPersistenceEntity have to have be in layer dataaccess
+     */
+    @ArchTest
+    private static final ArchRule DevonApplicationPersistenceEntityCheck = 
+            classes()
+                   .that().areAssignableTo(ApplicationPersistenceEntity.class).should().resideInAnyPackage("..dataaccess..")
+                   .because("Classes extending ApplicationPersistenceEntity must be located in the package dataaccess");
 
+    /**
+     * DevonAbstractUcCheck verifying that classes extending AbstractUc have to have be in layer logic
+     */
+    @ArchTest
+    private static final ArchRule DevonAbstractUcCheck = 
+            classes()
+                   .that().areAssignableTo(AbstractUc.class).should().resideInAnyPackage("..logic..")
+                   .because("Classes extending AbstractUc must be located in the package logic");
+       
+     /**
+     * DevonMapperCheck verifying that classes extending Mapper have to have be in layer logic and end with the suffix Mapper
+     */
+    @ArchTest
+    private static final ArchRule DevonMapperCheck = 
+            classes()   
+                   .that().areAnnotatedWith(Mapper.class).should().resideInAnyPackage("..logic..").andShould().haveSimpleNameEndingWith("Mapper")
+                   .because("Classes extending Mapper must be located in the package logic and end with Mapper");
```

Comment:
replace `Classes extending Mapper` with `Bean-Mappers` or `Types annotated with @Mapper` as our mappers are interfaces and not classes and they do not extend `Mapper` but are annotated with it.